scripts/cloudbuild.yaml

```diff
@@ -40,7 +40,24 @@ steps:
           echo "GLIBC_TAR_CACHE=false" >> /workspace/env.txt
         fi
 
-  # Step 3: Build the Docker image from the .devcontainer directory.
+  # Step 3: Detect the Git remote URL from the checked-out source.
+  # - Reads the remote.origin.url from the local .git config.
+  # - Writes the value to /workspace/repo_url.txt for later use in Docker build.
+  # - This supports building from forked repositories by preserving the correct source origin.
+
+  - name: 'gcr.io/cloud-builders/git'
+    id: 'Detect Repo URL'
+    entrypoint: 'bash'
+    args:
+      - '-c'
+      - |
+          cd /workspace
+          git config --get remote.origin.url > /workspace/repo_url.txt
+          echo "Detected REPO_URL:"
+          cat /workspace/repo_url.txt
```

Comment:
This seems overkill. Can't I use something like [`$REPO_FULL_NAME`](https://cloud.google.com/build/docs/configuring-builds/substitute-variable-values) below?

Reply:
Cloud Build reports REPO_FULL_NAME as the target, not the source, so it fails the same way as before

Reply:
Thanks for the insight!